CRM/Core/BAO/CustomQuery.php

```diff
@@ -387,20 +387,27 @@ public function where() {
 
               // CRM-14563,CRM-16575 : Special handling of multi-select custom fields
               if ($isSerialized && !empty($value) && !strstr($op, 'NULL') && !strstr($op, 'LIKE')) {
+                $sp = CRM_Core_DAO::VALUE_SEPARATOR;
                 if (strstr($op, 'IN')) {
-                  $value = str_replace(",", "[[:cntrl:]]*|[[:cntrl:]]*", $value);
+                  $value = str_replace(",", "$sp|$sp", $value);
                   $value = str_replace('(', '[[.left-parenthesis.]]', $value);
                   $value = str_replace(')', '[[.right-parenthesis.]]', $value);
                 }
                 $op = (strstr($op, '!') || strstr($op, 'NOT')) ? 'NOT RLIKE' : 'RLIKE';
-                $value = "[[:cntrl:]]*" . $value . "[[:cntrl:]]*";
+                $value = $sp . $value . $sp;
                 if (!$wildcard) {
```

Comment:
It shouldn't ... 
```mysql> select "A(B" regexp "[[.left-parenthesis.]]";
+---------------------------------------+
| "A(B" regexp "[[.left-parenthesis.]]" |
+---------------------------------------+
|                                     1 |
+---------------------------------------+

mysql> select "A(B" regexp "[[:cntrl:]]";
+----------------------------+
| "A(B" regexp "[[:cntrl:]]" |
+----------------------------+
|                          0 |
+----------------------------+```

But "(" in the regexp itself is special, hence replacing it with [[.left-parenthesis.]] before using the value in the regexp.